src/SessionDurationMetric.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+
+namespace Tightenco\NovaGoogleAnalytics;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Laravel\Nova\Metrics\Value;
+use Laravel\Nova\Metrics\ValueResult;
+use Spatie\Analytics\Analytics;
+use Spatie\Analytics\Period;
+
+class SessionDurationMetric extends Value
+{
+    public function name() {
+        return __('Avg. Session Duration');
+    }
+
+    /**
+     * Calculate the value of the metric.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return mixed
+     */
+    public function calculate(Request $request)
+    {
+        $lookups = [
+            1 => $this->sessionDurationOneDay(),
+            'MTD' => $this->sessionDurationOneMonth(),
+            'YTD' => $this->sessionDurationOneYear(),
+        ];
+
+        $data = Arr::get($lookups, $request->get('range'), ['result' => 0, 'previous' => 0]);
+
+        return (new ValueResult($data['result']))->previous($data['previous'])->format('00:00:00');
+    }
+
+    private function sessionDurationOneDay()
+    {
+        $analyticsData = app(Analytics::class)
+            ->performQuery(
+                Period::days(1),
+                'ga:avgSessionDuration',
+                [
+                    'metrics' => 'ga:avgSessionDuration',
+                    'dimensions' => 'ga:date',
+                ]
+            );
+
+        $results = collect($analyticsData->getRows());
+
+        return [
+            'result' => $results->last()[1] ?? 0,
+            'previous' => $results->first()[1] ?? 0,
+        ];
+    }
+
+    private function sessionDurationOneMonth()
+    {
+        $currentAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create(Carbon::today()->startOfMonth(), Carbon::today()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:yearMonth',
+            ]
+        );
+
+        $currentResults = collect($currentAnalyticsData->getRows());
+
+        $lastMonth = Carbon::today()->startOfMonth()->subMonths(1);
+        $previousAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create($lastMonth->startOfMonth(), $lastMonth->endOfMonth()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:yearMonth',
+            ]
+        );
+
+        $previousResults = collect($previousAnalyticsData->getRows());
+
+        return [
+            'previous' => $previousResults->last()[1] ?? 0,
+            'result' => $currentResults->last()[1] ?? 0,
+        ];
+    }
+
+    private function sessionDurationOneYear()
+    {
+        $currentAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create(Carbon::today()->startOfYear(), Carbon::today()),
```

Comment:
Is this correct for year? or should it be today sub year? It's always hard for me to know how that should be

Reply:
OK, so it looks like you're doing this year YTD compared to last year all year.

Should it instead be "the year that started one year ago and ends today, vs the year that started two years ago and ended one year ago"? I don't know what we're doing elsewhere

Reply:
We are doing Today, MTD, and YTD on the metric cards.

Reply:
OK, that answers the first half of the question.

But it opens a new question (yay!) lol:

If we are doing "YTD" on july 1, part of me thinks the user wants to compare
"This year january 1 to july 1" vs "last year january 1 to july 1"
But this code right here shows

"this year january 1 to july 1" vs "last year january 1 to december 31"

So... which do users expect?